generated/nise/session.py

```diff
@@ -0,0 +1,842 @@
+        disconnect_spec_ctype = ctypes.create_string_buffer(disconnect_spec.encode(self._encoding))  # case C020
+        error_code = self._library.niSE_Disconnect(vi_ctype, disconnect_spec_ctype)
+        errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
+        return
+
+    def disconnect_all(self):
+        '''disconnect_all
+
+        Disconnects all connections on every IVI switch device managed by the
+        NISE session reference passed to this method. disconnect_all
+        ignores all multiconnect modes. Calling disconnect_all resets all
+        of the switch states for the system.
+        '''
+        vi_ctype = _visatype.ViSession(self._vi)  # case S110
+        error_code = self._library.niSE_DisconnectAll(vi_ctype)
+        errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
+        return
+
+    def expand_route_spec(self, route_spec, expand_action=enums.ExpandAction.ROUTES, expanded_route_spec_size=[1024]):
+        '''expand_route_spec
+
+        Expands a route spec string to yield more information about the routes
+        and route groups within the spec. The route specification string
+        returned from expand_route_spec can be passed to other Switch
+        Executive API methods (such as connect, disconnect, and
+        connect_and_disconnect) that use route specification strings.
+
+        Args:
+            route_spec (str): String describing the routes and route groups to expand. The route
+                specification strings are best summarized as a series of routes
+                delimited by ampersands. The specified routes may be route names, route
+                group names, or fully specified route paths delimited by square
+                brackets. Some examples of route specification strings are: MyRoute
+                MyRouteGroup MyRoute & MyRouteGroup [A->Switch1/r0->B] MyRoute &
+                MyRouteGroup & [A->Switch1/r0->B] Refer to Route Specification Strings
+                in the NI Switch Executive Help for more information.
+
+            expand_action (enums.ExpandAction): This value sets the expand action for the method. The action might be
+                one of the following: NISE_VAL_EXPAND_TO_ROUTES (0) - expands the
+                route spec to routes. Converts route groups to their constituent routes.
+                NISE_VAL_EXPAND_TO_PATHS (1) - expands the route spec to paths.
+                Converts routes and route groups to their constituent square bracket
+                route spec strings. Example: [Dev1/c0->Dev1/r0->Dev1/c1]
+
+                Note:
+                One or more of the referenced values are not in the Python API for this driver. Enums that only define values, or represent True/False, have been removed.
+
+            expanded_route_spec_size (list of int): The routeSpecSize is an ViInt32 that is passed by reference into the
```

Comment:
* This is not a list of int
* This should not be exposed in the Python API

Reply:
In Switch Executive, for strings, you:
1. create a pointer 
2. Assign a value to that pointer with the size of the string buffer you have pre-allocated
3. Pass the pointer into the dll 
4. Read pointer containing the actual length of the string

This is similar to the IVI dance, except

1. The length of the string is not returned from the function. Switch Executive returns the error code always. If I used ivi-dance as a mechanism, then the string size is always 0, resulting in an empty string.
2. The direction is out instead of in

To resolve this for Switch Executive, I had to do the following.

- Change type from pointer to VIInt32 array of size 1 (that is essentially a pointer.)
- Set direction from out, to in (so both I and the dll could set the value, as not setting it would cause no string)
- Set default value of "Size" to the default used in examples,
- Use "Size" to allocated the size of the string buffer.

This enables the string to be read, and partially hidden from the customer (via defaults).

The other option is to create a completely new mechanism for filling buffers specific to Switch Executive.

Thoughts?

Reply:
I think this is what we should do. But I don't think that should hold up this PR, rather it should be a separate discussion altogether. Please create an issue about it.